addon/model.js

```diff
@@ -390,7 +398,8 @@ export default class MegamorphicModel extends EmberObject {
         // We update record arrays in-place to match the semantics of setting a
         // `hasMany` attribute on a `DS.Model`.
         this._setRecordArray(key, value);
-        notifyPropertyChange(this, key);
+        this._deferProprtyChange(key);
+        flushChanges(this._store);
```

Comment:
Why do we defer and then immediately flush?

Reply:
No reason; i'll change it to be explicitly immediate to better signify intent